src/api/app/controllers/webui/request_controller.rb

```diff
@@ -473,6 +479,10 @@ def set_actions
     @actions = @bs_request.bs_request_actions
   end
 
+  def set_build_results_data
+    @build_results_data = ActionBuildResultsService::ChartDataExtractor.new(actions: @actions).call
```

Comment:
You are using this function as a "getter" and a "setter" (to use JAVA parlance) at the same time. You should set the instance variable or use the return value of the function to set the locals above, but not both, please.

Reply:
I also thing `@build_results_data` is not used anywhere...

Reply:
Good catch. It's not used anymore now. I'm refactoring it :+1:

Reply:
done